### PR TITLE
update paper data example

### PR DESCRIPTION
**Pull request by manonreau**
Add examples of code to run the CNN models from the paper on new data:

- Example with 3DeepFace
- Example with the docking scoring model

- Changed NeuralNet and DataSet to handle test set with no target value

### CODE REVIEW

File: deeprank/learn/NeuralNet.py [hunk 1]
```diff
@@ -494,7 +495,10 @@ def load_nn_params(self):
         """Get NeuralNet parameters from a saved model."""
         self.task = self.state['task']
         self.criterion = self.state['criterion']
-        self.hit_cutoff = self.state['hit_cutoff']
+        try:
+            self.hit_cutoff = self.state['hit_cutoff']
+        except Exception:
+            print('No hit_cutoff provided')
```
Comment (CunliangGeng) [inline deeprank/learn/NeuralNet.py:501]: Please use `logger` to output messages. 
I guess this `Exception` only applies to our published models. If so, we have to tell users what to do in the message, like `No "hit_cutoff" found in {model}. Please set it in function when doing prediction"`.

File: deeprank/learn/NeuralNet.py [hunk 2]
```diff
@@ -766,7 +770,8 @@ def _train(self, index_train, index_valid, index_test,
         return torch.cat([param.data.view(-1)
                           for param in self.net.parameters()], 0)
 
-    def _epoch(self, data_loader, train_model):
+
+    def _epoch(self, data_loader, train_model, test_model=False):
```
Comment (CunliangGeng) [inline deeprank/learn/NeuralNet.py:774]: The name `test_model` is misleading. It's acutally a flag to tell whether there is `target` values or not in the data. So I suggest using e.g. `has_target=True`?
Also, please add the docstring for new parameters.

File: deeprank/learn/DataSet.py [hunk 3]
```diff
@@ -1217,6 +1218,7 @@ def map_one_molecule(self, fname, mol=None, angle=None, axis=None):
         try:
             target = mol_data.get('targets/' + self.select_target)[()]
         except Exception:
+            target = None
             logger.exception(f'No target value for: {fname} - not required for the test set')
```
Comment (CunliangGeng) [inline deeprank/learn/DataSet.py:1222]: Make more sense to change `logger.exception` to `logger.warning`?

File: deeprank/learn/DataSet.py [hunk 4]
```diff
@@ -1168,6 +1168,7 @@ def load_one_molecule(self, fname, mol=None):
...
```
Comment (CunliangGeng) [inline deeprank/learn/DataSet.py:1172]: Make more sense to change logger.exception to logger.warning?

**Review by CunliangGeng — CHANGES_REQUESTED** [review]
Thanks @manonreau I left a few comments for you to check

File: deeprank/learn/DataSet.py (hunk 4 above)
Comment (manonreau) [reply deeprank/learn/DataSet.py]: Thanks Liang, I agree.

File: deeprank/learn/DataSet.py (hunk 3 above)
Comment (manonreau) [reply deeprank/learn/DataSet.py]: Same :)

File: deeprank/learn/NeuralNet.py (hunk 1 above)
Comment (manonreau) [reply deeprank/learn/NeuralNet.py]: The hit cutoff is used only if target values are provided. In prediction mode you do not need it, you need it for benchmark only.

File: deeprank/learn/NeuralNet.py (hunk 2 above)
Comment (manonreau) [reply deeprank/learn/NeuralNet.py]: Very good suggestion, thanks !

**Review by manonreau — COMMENTED** [review]
Thanks Liang for the review, I modified the code accordingly.

File: deeprank/learn/NeuralNet.py [hunk 5]
```diff
@@ -494,7 +499,10 @@ def load_nn_params(self):
         """Get NeuralNet parameters from a saved model."""
         self.task = self.state['task']
         self.criterion = self.state['criterion']
-        self.hit_cutoff = self.state['hit_cutoff']
+        try:
+            self.hit_cutoff = self.state['hit_cutoff']
+        except Exception:
+            print(f'No "hit_cutoff" found in {self.pretrained_model}. Please set it in function "test()" when doing benchmark"')
```
Comment (CunliangGeng) [inline deeprank/learn/NeuralNet.py:505]: Please replace `print` with `logger` :-)

**Review by CunliangGeng — APPROVED** [review]
Thanks @manonreau , the changes look good!
One reminder for you is that we don't use `print` but `logger` for outputing message in DeepRank :-)

File: deeprank/learn/NeuralNet.py (hunk 5 above)
Comment (manonreau) [reply deeprank/learn/NeuralNet.py]: done
